src/gluonts/dataset/parallelized_loader.py

```diff
@@ -239,29 +239,16 @@ def _sequential_sample_generator(
     cyclic: bool,
     num_batches_for_shuffling: int,
 ) -> Iterator[DataEntry]:
-    # Approximate shuffling `num_batches_for_shuffling` probabilistically
-    def skip_data_iter(data: Dataset):
-        for data_entry in data:
-            if (
-                random.randint(1, num_batches_for_shuffling)
-                == num_batches_for_shuffling
-            ):
-                yield data_entry
-
     # sanity check: since `num_batches_for_shuffling` works by skipping
     # entries, this should not be used for non cyclic datasets
     assert (
         cyclic or num_batches_for_shuffling == 1
     ), "Setting num_batches_for_shuffling >= 1 only makes sense in the context of cyclic datasets currently."
 
     while True:
-        for sample in transformation(
-            data_it=skip_data_iter(dataset)
-            if num_batches_for_shuffling > 1
-            else dataset,
-            is_train=is_train,
-        ):
-            yield sample
```

Comment:
So I think this was put here to have a way of shuffling an iterable: this change removes this completely I suppose? If so, `num_batches_for_shuffling` is maybe not required anymore

Reply:
Yes, let's maybe remove this for now. There are some issues with performance when using this and I think our assumption has always been training datasets should be shuffled. I think we could reintroduce this in a simpler way, by keeping a longer batch of data and shuffling this.

Reply:
Yes, we also saw some performance issues with this in MQCNN, the read speeds are just to slow. Greater batch size should work.